xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
All I wanted to say is that a lock or atomic won't change anything in current behaviour but I am not against introducing an atomic per se. ThreadIdentifier is maybe not a good candidate for an atomic.